Add dl linking to gio tests to avoid:
gio/tests/socket-listener.p/socket-listener.c.o: undefined reference to symbol 'dlsym@@GLIBC_2.2.5'

Patch added by Simon Branford (University of Birmingham)
--- gio/tests/meson.build.orig	2025-06-20 13:15:21.974161000 +0100
+++ gio/tests/meson.build	2025-06-20 13:15:40.689118893 +0100
@@ -1,4 +1,5 @@
 common_gio_tests_deps = [
+  libdl_dep,
   libglib_dep,
   libgmodule_dep,
   libgobject_dep,
